config/interpolate_funcs.go

```diff
@@ -1548,6 +1549,50 @@ func interpolationFuncURLEncode() ast.Function {
 	}
 }
 
+// interpolationFuncTranspose implements the "transpose" function
+// that converts a map (string,list) to a map (string,list) where
+// the unique values of the original lists become the keys of the
+// new map and the keys of the original map become values for the
+// corresponding new keys.
+func interpolationFuncTranspose() ast.Function {
+	return ast.Function{
+		ArgTypes:   []ast.Type{ast.TypeMap},
+		ReturnType: ast.TypeMap,
+		Callback: func(args []interface{}) (interface{}, error) {
+
+			inputMap := args[0].(map[string]ast.Variable)
+			outputMap := make(map[string]ast.Variable)
+			tmpMap := make(map[string][]string)
+
+			for inKey, inVal := range inputMap {
+				values := inVal.Value.([]ast.Variable)
+				for _, listVal := range values {
+					if listVal.Type != ast.TypeString {
+						return nil, fmt.Errorf("lists of (%s) are not supported", listVal.Type.Printable())
```

Comment:
This is a good thing to check here, for the same reasons as above. I'd suggest a more "actionable" error message: tell the user what is required, rather than what's _disallowed_:

```go
  return nil, fmt.Errorf("transpose requires the given map values to be lists of strings")
```